app/src/main/java/dev/arkbuilders/arkmemo/ui/views/SwitchButton.kt

```diff
@@ -0,0 +1,883 @@
+        uncheckCircleWidth = optPixelSize(
+            typedArray,
+            R.styleable.SwitchButton_sb_uncheckcircle_width,
+            dp2pxInt(1.5f)
+        ) //dp2pxInt(1.5f);
+        uncheckCircleOffsetX = 10f.dp2Px()
+        uncheckCircleRadius = optPixelSize(
+            typedArray,
+            R.styleable.SwitchButton_sb_uncheckcircle_radius,
+            4f.dp2Px()
+        ) //dp2px(4);
+        checkedLineOffsetX = 4f.dp2Px()
+        checkedLineOffsetY = 4f.dp2Px()
+        shadowRadius = optPixelSize(
+            typedArray,
+            R.styleable.SwitchButton_sb_shadow_radius,
+            dp2pxInt(2.5f)
+        ) //dp2pxInt(2.5f);
+        shadowOffset = optPixelSize(
+            typedArray,
+            R.styleable.SwitchButton_sb_shadow_offset,
+            dp2pxInt(1.5f)
+        ) //dp2pxInt(1.5f);
+        shadowColor = optColor(
+            typedArray,
+            R.styleable.SwitchButton_sb_shadow_color,
+            0X33000000
+        ) //0X33000000;
+        uncheckColor = optColor(
+            typedArray,
+            R.styleable.SwitchButton_sb_uncheck_color,
+            -0x222223
+        ) //0XffDDDDDD;
+        checkedColor = optColor(
+            typedArray,
+            R.styleable.SwitchButton_sb_checked_color,
+            -0xae2c99
+        ) //0Xff51d367;
+        borderWidth = optPixelSize(
+            typedArray,
+            R.styleable.SwitchButton_sb_border_width,
+            dp2pxInt(1f)
+        ) //dp2pxInt(1);
+        checkLineColor = optColor(
+            typedArray,
+            R.styleable.SwitchButton_sb_checkline_color,
+            Color.WHITE
+        ) //Color.WHITE;
+        checkLineWidth = optPixelSize(
+            typedArray,
+            R.styleable.SwitchButton_sb_checkline_width,
+            dp2pxInt(1f)
+        ) //dp2pxInt(1.0f);
+        checkLineLength = 6f.dp2Px()
+        val buttonColor = optColor(
+            typedArray,
+            R.styleable.SwitchButton_sb_button_color,
+            Color.WHITE
+        ) //Color.WHITE;
+        uncheckButtonColor = optColor(
+            typedArray,
+            R.styleable.SwitchButton_sb_uncheckbutton_color,
+            buttonColor
+        )
+        checkedButtonColor = optColor(
+            typedArray,
+            R.styleable.SwitchButton_sb_checkedbutton_color,
+            buttonColor
+        )
+        val effectDuration = optInt(
+            typedArray,
+            R.styleable.SwitchButton_sb_effect_duration,
+            300
+        ) //300;
+        isChecked = optBoolean(
+            typedArray,
+            R.styleable.SwitchButton_sb_checked,
+            false
+        )
+        showIndicator = optBoolean(
+            typedArray,
+            R.styleable.SwitchButton_sb_show_indicator,
+            true
+        )
+        background = optColor(
+            typedArray,
+            R.styleable.SwitchButton_sb_background,
+            Color.WHITE
+        ) //Color.WHITE;
```

Comment:
Maybe we could have a hint of why we use 300
```suggestion
)
```

Reply:
@shuberttheI I have removed the useless comment.